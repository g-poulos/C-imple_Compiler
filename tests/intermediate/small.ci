program small
{
    # declare A:=1; #
    declare b,g,f;
    function P1(in X, inout Y)
    { declare e,f;
        function P11(inout X)
        {   declare e;
            e:=A;
            X:=Y;
            f:=b;
            return(e);
        }

        # code for P1 #
        b:=X;
        e:=P11(inout X);
        e:=P1(in X,inout Y);
        X:=b;
        return(e);
    }
    # code for main #
    if (b>1 and f<2 or g+1<f+b)
    {
        f:=P1(in g);
    }
    else
    {
        f:=1;
    }
}.
