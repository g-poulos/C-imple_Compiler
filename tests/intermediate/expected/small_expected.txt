1: begin_block P11 _ _
2: := A _ e
3: := Y _ X
4: := b _ f
5: RET e _ _
6: end_block P11 _ _
7: begin_block P1 _ _
8: := X _ b
9: par X REF _
10: par T_0 RET _
11: call P11 _ _
12: := T_0 _ e
13: par X CV _
14: par Y REF _
15: par T_1 RET _
16: call P1 _ _
17: := T_1 _ e
18: := b _ X
19: RET e _ _
20: end_block P1 _ _
21: begin_block small _ _
22: > b 1 24
23: jump _ _ 26
24: < f 2 30
25: jump _ _ 26
26: + g 1 T_2
27: + f b T_3
28: < T_2 T_3 30
29: jump _ _ 35
30: par g CV _
31: par T_4 RET _
32: call P1 _ _
33: := T_4 _ f
34: jump _ _ 36
35: := 1 _ f
36: halt _ _ _
37: end_block small _ _